api/v1alpha1/contour_types.go

```diff
@@ -280,6 +302,32 @@ const (
 	GCPLoadBalancerProvider   LoadBalancerProviderType = "GCP"
 )
 
+// NodePort is the schema to specify a network port for a NodePort Service.
+type NodePort struct {
+	// Name is an IANA_SVC_NAME within the NodePort Service.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
```

Comment:
To make sure the port names are `http` or `https` could we add kubebuilder validation here? or do we want to keep the logic all in the operator itself

Reply:
We could use `+kubebuilder:validation:Enum` to enforce "http" and "https" if `name` was a custom type. I think we want to stick with a string so the types doesn't need to change when Contour supports more than 2 listeners.

Reply:
we use the `Enum` validation on string fields in Contour, does that not work? See  https://github.com/projectcontour/contour/blob/main/apis/projectcontour/v1/httpproxy.go#L371 https://github.com/projectcontour/contour/blob/main/apis/projectcontour/v1/httpproxy.go#L542